From: Pu Kong <pukong.wjm@taobao.com>
Subject: [PATCH 1/2] export symbols needed for TOA kernel module
Patch-mainline: in-house-usage

This patch exports several symbols which is needed by TOA kernel module.
TOA: Tcp Option of Address.

Signed-off-by: Pu Kong <pukong.wjm@taobao.com>
Signed-off-by: Coly Li <bosong.ly@taobao.com>
---
Index: linux-2.6.32-220.2.1.el5/include/net/inet_common.h
===================================================================
--- linux-2.6.32-220.2.1.el5.orig/include/net/inet_common.h	2012-02-15 15:12:40.570697374 +0800
+++ linux-2.6.32-220.2.1.el5/include/net/inet_common.h	2012-02-17 13:17:18.996644302 +0800
@@ -1,7 +1,7 @@
 #ifndef _INET_COMMON_H
 #define _INET_COMMON_H
 
-extern const struct proto_ops		inet_stream_ops;
+extern struct proto_ops			inet_stream_ops;
 extern const struct proto_ops		inet_dgram_ops;
 
 /*
Index: linux-2.6.32-220.2.1.el5/include/net/ipv6.h
===================================================================
--- linux-2.6.32-220.2.1.el5.orig/include/net/ipv6.h	2012-02-15 15:12:40.574697401 +0800
+++ linux-2.6.32-220.2.1.el5/include/net/ipv6.h	2012-02-17 13:58:31.808905997 +0800
@@ -598,7 +598,7 @@
 /*
  * reassembly.c
  */
-extern const struct proto_ops inet6_stream_ops;
+const struct proto_ops inet6_stream_ops;
 extern const struct proto_ops inet6_dgram_ops;
 
 struct group_source_req;
@@ -648,5 +648,11 @@
 extern void ipv6_static_sysctl_unregister(void);
 #endif
 
+/* public func in tcp_ipv6.c */
+extern struct sock *tcp_v6_syn_recv_sock(struct sock *sk, struct sk_buff *skb,
+					struct request_sock *req,
+					struct dst_entry *dst);
+extern const struct inet_connection_sock_af_ops ipv6_specific;
+
 #endif /* __KERNEL__ */
 #endif /* _NET_IPV6_H */
Index: linux-2.6.32-220.2.1.el5/include/net/transp_v6.h
===================================================================
--- linux-2.6.32-220.2.1.el5.orig/include/net/transp_v6.h	2012-02-15 15:12:40.570697374 +0800
+++ linux-2.6.32-220.2.1.el5/include/net/transp_v6.h	2012-02-17 13:17:19.000644323 +0800
@@ -53,7 +53,7 @@
 /*
  *	address family specific functions
  */
-extern const struct inet_connection_sock_af_ops ipv4_specific;
+extern struct inet_connection_sock_af_ops ipv4_specific;
 
 extern void inet6_destroy_sock(struct sock *sk);
 
Index: linux-2.6.32-220.2.1.el5/kernel/kallsyms.c
===================================================================
--- linux-2.6.32-220.2.1.el5.orig/kernel/kallsyms.c	2012-02-15 15:12:41.254700765 +0800
+++ linux-2.6.32-220.2.1.el5/kernel/kallsyms.c	2012-02-17 13:17:19.000644323 +0800
@@ -181,6 +181,7 @@
 	}
 	return module_kallsyms_lookup_name(name);
 }
+EXPORT_SYMBOL_GPL(kallsyms_lookup_name);
 
 int kallsyms_on_each_symbol(int (*fn)(void *, const char *, struct module *,
 				      unsigned long),
Index: linux-2.6.32-220.2.1.el5/net/ipv4/af_inet.c
===================================================================
--- linux-2.6.32-220.2.1.el5.orig/net/ipv4/af_inet.c	2012-02-15 15:12:41.142700207 +0800
+++ linux-2.6.32-220.2.1.el5/net/ipv4/af_inet.c	2012-02-17 13:17:19.000644323 +0800
@@ -871,7 +871,7 @@
 }
 EXPORT_SYMBOL(inet_ioctl);
 
-const struct proto_ops inet_stream_ops = {
+struct proto_ops inet_stream_ops = {
 	.family		   = PF_INET,
 	.owner		   = THIS_MODULE,
 	.release	   = inet_release,
Index: linux-2.6.32-220.2.1.el5/net/ipv4/tcp_ipv4.c
===================================================================
--- linux-2.6.32-220.2.1.el5.orig/net/ipv4/tcp_ipv4.c	2012-02-17 13:17:18.248640596 +0800
+++ linux-2.6.32-220.2.1.el5/net/ipv4/tcp_ipv4.c	2012-02-17 13:17:19.000644323 +0800
@@ -1762,7 +1762,7 @@
 	return 0;
 }
 
-const struct inet_connection_sock_af_ops ipv4_specific = {
+struct inet_connection_sock_af_ops ipv4_specific = {
 	.queue_xmit	   = ip_queue_xmit,
 	.send_check	   = tcp_v4_send_check,
 	.rebuild_header	   = inet_sk_rebuild_header,
Index: linux-2.6.32-220.2.1.el5/net/ipv6/af_inet6.c
===================================================================
--- linux-2.6.32-220.2.1.el5.orig/net/ipv6/af_inet6.c	2012-02-15 15:12:41.138700186 +0800
+++ linux-2.6.32-220.2.1.el5/net/ipv6/af_inet6.c	2012-02-17 13:17:19.000644323 +0800
@@ -503,7 +503,7 @@
 
 EXPORT_SYMBOL(inet6_ioctl);
 
-const struct proto_ops inet6_stream_ops = {
+struct proto_ops inet6_stream_ops = {
 	.family		   = PF_INET6,
 	.owner		   = THIS_MODULE,
 	.release	   = inet6_release,
@@ -528,6 +528,7 @@
 	.compat_getsockopt = compat_sock_common_getsockopt,
 #endif
 };
+EXPORT_SYMBOL(inet6_stream_ops);
 
 const struct proto_ops inet6_dgram_ops = {
 	.family		   = PF_INET6,
Index: linux-2.6.32-220.2.1.el5/net/ipv6/tcp_ipv6.c
===================================================================
--- linux-2.6.32-220.2.1.el5.orig/net/ipv6/tcp_ipv6.c	2012-02-15 15:12:41.138700186 +0800
+++ linux-2.6.32-220.2.1.el5/net/ipv6/tcp_ipv6.c	2012-02-17 13:58:04.956773061 +0800
@@ -77,7 +77,7 @@
 static int	tcp_v6_do_rcv(struct sock *sk, struct sk_buff *skb);
 
 static const struct inet_connection_sock_af_ops ipv6_mapped;
-static const struct inet_connection_sock_af_ops ipv6_specific;
+const struct inet_connection_sock_af_ops ipv6_specific;
 #ifdef CONFIG_TCP_MD5SIG
 static const struct tcp_sock_af_ops tcp_sock_ipv6_specific;
 static const struct tcp_sock_af_ops tcp_sock_ipv6_mapped_specific;
@@ -1257,7 +1257,7 @@
 	return 0; /* don't send reset */
 }
 
-static struct sock * tcp_v6_syn_recv_sock(struct sock *sk, struct sk_buff *skb,
+struct sock *tcp_v6_syn_recv_sock(struct sock *sk, struct sk_buff *skb,
 					  struct request_sock *req,
 					  struct dst_entry *dst)
 {
@@ -1467,6 +1467,7 @@
 	NET_INC_STATS_BH(sock_net(sk), LINUX_MIB_LISTENDROPS);
 	return NULL;
 }
+EXPORT_SYMBOL(tcp_v6_syn_recv_sock);
 
 static __sum16 tcp_v6_checksum_init(struct sk_buff *skb)
 {
@@ -1768,7 +1769,7 @@
 	return 0;
 }
 
-static const struct inet_connection_sock_af_ops ipv6_specific = {
+const struct inet_connection_sock_af_ops ipv6_specific = {
 	.queue_xmit	   = inet6_csk_xmit,
 	.send_check	   = tcp_v6_send_check,
 	.rebuild_header	   = inet6_sk_rebuild_header,
@@ -1786,6 +1787,7 @@
 	.compat_getsockopt = compat_ipv6_getsockopt,
 #endif
 };
+EXPORT_SYMBOL(ipv6_specific);
 
 #ifdef CONFIG_TCP_MD5SIG
 static const struct tcp_sock_af_ops tcp_sock_ipv6_specific = {
